### PR TITLE
Feature/blui 4135 eula screen base

### PR DESCRIPTION
**Pull request by ektaghag-eaton**
Fixes 4135 .

#### Changes proposed in this Pull Request:

- Eula Screen implementation
- Updated eulaContent type as `string | JSX element`
- add one more prop i.e. CheckboxProps to use all props of MUI Checkbox

#### Screenshots / Screen Recording (if applicable)

-

#### To Test:

-http://localhost:3000/eula-screen-test

#### Any specific feedback you are looking for?

-

### CODE REVIEW

**Comment by manojleaton** [comment]
There is more space between checkbox and the divider, Please reduce that
Screenshot 2023-06-01 at 11 47 24 AM

File: login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ *
+ * @param eulaContent the content to render for the EULA. Can be a plain string or HTML
+ * @param htmlEula true if the EULA should be rendered as HTML
+ * @param checkboxLabel label for the EULA checkbox
+ * @param initialCheckboxValue used to pre-populate the checked/unchecked checkbox when the screen loads
+ * @param checkboxProps used to set checkbox props
+ * @category Component
+ */
+
+export const EulaScreenBase: React.FC<React.PropsWithChildren<EulaScreenProps>> = (props) => {
```
Comment (manojleaton) [inline login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx:24]: The `React.PropsWithChildren` is not needed here, as you are not using the children prop explicitly. You can simply use `EulaScreenProps`

File: login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const handleEulaAcceptedChecked = useCallback(
+        (accepted: boolean) => {
+            setEulaAccepted(onEulaAcceptedChange(accepted));
+        },
+        [onEulaAcceptedChange]
+    );
+
+    return (
+        <WorkflowCard {...cardBaseProps}>
+            <WorkflowCardHeader {...headerProps}></WorkflowCardHeader>
```
Comment (manojleaton) [inline login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx:49]: You can use the WorkflowCardHeader component as self closing element, as we are not passing any children to it.   
```suggestion
            <WorkflowCardHeader {...headerProps} />
```

**Comment by ektaghag-eaton** [comment]
> There is more space between checkbox and the divider, Please reduce that
> 
> Screenshot 2023-06-01 at 11 47 24 AM

Hi @manojleaton , need to debug more on this issue. Extra space comes from the WorkflowCardContent component. If other things are fine, I will raise an issue for this and merge this PR.

File: login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+/** eslint-ignore */
+import React, { useCallback, useState } from 'react';
+import { EulaScreenProps } from './types';
+import { WorkflowCard } from '../../components/WorkflowCard';
+import { WorkflowCardActions } from '../../components/WorkflowCard/WorkflowCardActions';
+import { WorkflowCardBody } from '../../components/WorkflowCard/WorkflowCardBody';
+import { WorkflowCardHeader } from '../../components/WorkflowCard/WorkflowCardHeader';
```
Comment (manojleaton) [inline login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx:7]: You can simplified the imports like this
`import { WorkflowCard, WorkflowCardActions, WorkflowCardBody, WorkflowCardHeader } from '../../components/WorkflowCard';
`

Comment (ektaghag-eaton) [reply login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx]: Done

**Comment by ektaghag-eaton** [comment]
Hi @manojleaton , raised an issue [298](https://github.com/etn-ccis/blui-react-workflows/issues/298) here. Please approve if other things are fine.

File: login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
+/** eslint-ignore */
+import React, { useCallback, useState } from 'react';
+import { EulaScreenProps } from './types';
+import { WorkflowCard, WorkflowCardActions, WorkflowCardBody, WorkflowCardHeader } from '../../components/WorkflowCard';
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx:4]: No need to import it from WorkflowCard, importing just from '../../components' will work, as we exported those from components.

**Review by surajeaton — APPROVED** [review]
Looks good.

**Pull request merged**